WalletWasabi.Tests/IntegrationTests/RegTests.cs

```diff
@@ -238,6 +238,68 @@ public async Task FilterBuilderTestAsync()
 			}
 		}
 
+		[Fact]
+		public async Task StatusRequestTestAsync()
+		{
+			const string request = "/api/v3/btc/Blockchain/status";
+			(string password, RPCClient rpc, Network network, Coordinator coordinator, ServiceConfiguration serviceConfiguration, BitcoinStore bitcoinStore, Backend.Global global) = await InitializeTestEnvironmentAsync(1);
+
+			var indexBuilderService = global.IndexBuilderService;
+			try
+			{
+				indexBuilderService.Synchronize();
+				// Test initial synchronization.
+				var times = 0;
+				uint256 firstHash = await rpc.GetBlockHashAsync(0);
+				while (indexBuilderService.GetFilterLinesExcluding(firstHash, 101, out _).filters.Count() != 101)
+				{
+					if (times > 500) // 30 sec
+					{
+						throw new TimeoutException($"{nameof(IndexBuilderService)} test timed out.");
+					}
+					await Task.Delay(100);
+					times++;
+				}
+
+				using var client = new WasabiClient(new Uri(RegTestFixture.BackendEndPoint), null);
+				var response = await client.TorClient.SendAndRetryAsync(HttpMethod.Get, HttpStatusCode.OK, request);
+				using (HttpContent content = response.Content)
+				{
+					var resp = await content.ReadAsJsonAsync<StatusResponse>();
+					Assert.True(resp.FilterCreationActive);
+				}
+
+				// Simulate an unintended stop
+				await indexBuilderService.StopAsync();
+				indexBuilderService = null;
+
+				await rpc.GenerateAsync(1);
+
+				response = await client.TorClient.SendAndRetryAsync(HttpMethod.Get, HttpStatusCode.OK, request);
+				using (HttpContent content = response.Content)
+				{
+					var resp = await content.ReadAsJsonAsync<StatusResponse>();
+					Assert.True(resp.FilterCreationActive);
+				}
+
+				await rpc.GenerateAsync(1);
+
+				response = await client.TorClient.SendAndRetryAsync(HttpMethod.Get, HttpStatusCode.OK, request);
+				using (HttpContent content = response.Content)
+				{
+					var resp = await content.ReadAsJsonAsync<StatusResponse>();
+					Assert.False(resp.FilterCreationActive);
```

Comment:
It is failing here. IndexBuilderService stopped so it is not invalidating the cache. Noob solution to `Task.Delay()` here until the cache invalidated. We need to find a general solution to solve this because caching is good but makes tests fail.

Reply:
I solved it.